hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/logical/LogicalRules.java

```diff
@@ -40,6 +41,7 @@ public static RuleSet getRuleSet() {
                 FilterMergeRule.INSTANCE,
                 FilterProjectTransposeRule.INSTANCE,
                 FilterIntoScanLogicalRule.INSTANCE,
+                ReduceExpressionsRule.FILTER_INSTANCE,
```

Comment:
Shouldn't we also add PROJECT_INSTANCE and JOIN_INSTANCE?

Reply:
`JOIN_INSTANCE` will be included with joins.
Regarding `PROJECT_INSTANCE` I could not find a case where it would be applicable apart from `CASE` which is not supported (yet).

Reply:
JOIN_INSTANCE - my mistake

PROJECT_INSTANCE: I thought it will simplify `a + 1 + 1` to `a + 2`, but it doesn't... Also that `a IS NULL AND a IS NOT NULL` is simplified to `false`, but it doesn't... I thought Calcite does this.